lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_method.rb

```diff
@@ -212,5 +213,42 @@ def self.cleanup(method_pid, threads)
       threads.each(&:exit)
     end
     private_class_method :cleanup
+
+    def self.get_bodies_and_line_numbers(obj, aem)
+      embeds = get_embeds(obj.workspace, aem) << {:data => aem.data, :fqname => aem.fqname}
+      code_start = 0
+      line_hash = {}
+      bodies = []
+      embeds.each do |item|
+        bodies << item[:data]
+        line_count = item[:data].lines.count
+        code_end   = code_start + line_count
+        line_hash[item[:fqname]] = {:start => code_start, :end => code_end}
+        code_start = code_end + 1
+      end
+      return bodies, line_hash
+    end
+    private_class_method :get_bodies_and_line_numbers
+
+    def self.get_embeds(workspace, method_obj)
+      method_obj.embedded_methods.collect do |name|
+        method_name, klass, ns = get_embed_method_name(name)
+        match_ns = workspace.overlay_method(ns, klass, method_name)
+        cls = ::MiqAeClass.find_by_fqname("#{match_ns}/#{klass}")
+        aem = ::MiqAeMethod.find_by(:class_id => cls.id, :name => method_name) if cls
+        raise MiqAeException::MethodNotFound, "Embedded method #{name} not found" unless aem
+        fqname = "#{match_ns}/#{klass}/#{method_name}"
```

Comment:
If it did, I think it would consolidate the last four lines into:

```ruby
aem = ::MiqAeMethod.find_by_fqname("#{match_ns}/#{klass}/#{method_name}")
raise MiqAeException::MethodNotFound, "Embedded method #{name} not found" unless aem
```